drivers/console/rtt_console.c

```diff
@@ -18,14 +18,50 @@
 extern void __printk_hook_install(int (*fn)(int));
 extern void __stdout_hook_install(int (*fn)(int));
 
+static bool host_present = false;
+
 static int rtt_console_out(int character)
 {
 	unsigned int key;
 	char c = (char)character;
+	unsigned cnt;
+	int watchdog_counter = CONFIG_RTT_TX_RETRY_CNT;
+
+	do {
+		key = irq_lock();
+		cnt = SEGGER_RTT_WriteNoLock(0, &c, 1);
+		irq_unlock(key);
+
+		/* There are two possible reasons for not writing any data to RTT:
+		 * - The host is not connected and not reading the data.
+		 * - The buffer got full and will be read by the host.
+		 * These two situations are distinguished using the following
+		 * algorithm:
+		 * At the beginning, the module assumes that the host is active,
+		 * so when no data is read, it busy waits and retries.
+		 * If, after retrying, the host reads the data, the module
+		 * assumes that the host is active. If it fails, the module
+		 * assumes that the host is inactive and stores that
+		 * information. On next call, only one attempt takes place.
+		 * The host is marked as active if the attempt is successful.
+		 */
+		if (cnt == 0) {
+			if (host_present) {
```

Comment:
Is it possible to get information of host connection just form the RTT peripheral? (I ques not)

Reply:
it's not possible. The only indication is that buffer gets emptied.

Reply:
Maybe reach out to segger to request this as a new feature?